Automattic/portkey-android
# [Compile Warnings As Errors] Resolve Warnings & Enable All Warnings as Errors

This PR resolves/suppresses all warnings for all modules and then enables all warnings as errors as the default for this project (a976caf9c721c7d54c8f4c9e58ca4d92c5af48de).

PS: This is just a follow-up PR of the previous more important #734 PR to restrict further Kotlin related warnings to creep into this repo and with it to also enable future Kotlin updates, for example see below warning that would have made the Kotlin `1.7` update a bit more in the future:

`Non exhaustive 'when' statements on enum will be prohibited in 1.7, add 'XYZ' branches or 'else' branch instead`

-----

Warnings Resolution List:

- `mp4compose`:
    1. [Resolve unnecessary assertion type warning for mp4compose](https://github.com/Automattic/stories-android/commit/53a0c011f2cfec508406f99cdca1ee19cc32b99f)
    2. [Resolve unnecessary safe call type warning for mp4compose](https://github.com/Automattic/stories-android/commit/f1ebd698131594e06379aec7751f2b41b1c0eed8)
    3. [Resolve string type mismatch type warnings for mp4compose](https://github.com/Automattic/stories-android/commit/e9dfdee280006b8c4a7c9f6d2f0161c965d9a454)
    4. [Resolve byte array type mismatch type warnings for mp4compose](https://github.com/Automattic/stories-android/commit/9d01c416ec8827544bf9ba03db4d821b0f1d0df6)
    5. [Resolve variable is never used warnings for mp4compose](https://github.com/Automattic/stories-android/commit/d50209563e3200c844bd9dd0037b48902c83345d)
- `photoeditor`
    1. [Resolve non exhaustive when warnings for photoeditor](https://github.com/Automattic/stories-android/commit/21979a84ad576c994e562aa28ae28a8f80445aca)
    2. [Resolve parameter is never used warning for photoeditor](https://github.com/Automattic/stories-android/commit/731a66b9bec8034f635f043ef57cc9dfb2afc052)
    3. [Resolve parameter be renamed to _ warnings for photoeditor](https://github.com/Automattic/stories-android/commit/b0d2a60f7b2886473e883ca93fed488259fecd63)
    4. [Resolve string type mismatch type warnings for photoeditor](https://github.com/Automattic/stories-android/commit/d8936c0256873922e264e03b95b9b87eb9b3c776)
- `stories`
    1. [Resolve non exhaustive when warning for stories](https://github.com/Automattic/stories-android/commit/85d40e3e5496d5c4afc3358af860b4286cddc3fe)
    2. [Resolve condition always true warning for stories](https://github.com/Automattic/stories-android/commit/73e3ac8197b0e9e02f0bbd25530388538daf3d9d)
    3. [Resolve to lower case deprecated warnings for stories](https://github.com/Automattic/stories-android/commit/5b6aa06e5e98840916a35b3ccc0629dc38a18cc2)
- `app`
    1. [Resolve parameter be renamed to _ warnings for app](https://github.com/Automattic/stories-android/commit/da03525a0516d665c7d2c2bf6c65e12dafff81e9)

Warnings Suppression List:

- `mp4compose`
    1. [Suppress all deprecated warnings for mp4compose](https://github.com/Automattic/stories-android/commit/3403c2ad6f9a158adf93fd3adc31ec95571f4a32)
- `photoeditor`
    1. [Suppress listener deprecated warnings for photoeditor](https://github.com/Automattic/stories-android/commit/5cbbb01b1170dda3cd0a86eceeb93c309647fcfe)
    2. [Suppress parameter is never used warning for photoeditor](https://github.com/Automattic/stories-android/commit/77bc1e44839ce95998dafd583005033ad1e82399)
    3. [Suppress all deprecated warnings for photoeditor](https://github.com/Automattic/stories-android/commit/20a53ebf2199f23e52d875a6e8529681925a0347)
- `stories`
    1. [Suppress all deprecated warnings for stories](https://github.com/Automattic/stories-android/commit/8225f3c51e6ac3987fca8057c6abae090197ac8f)
    2. [Suppress parameter is never used warning for stories](https://github.com/Automattic/stories-android/commit/667b2d50dfce872d85703c33e54619424bb8b6f6)
- `app`
    1. [Suppress all deprecated warnings for app](https://github.com/Automattic/stories-android/commit/ba91f2532cc0119937de0f2f74ec8add43b69ef6)

-----

## To test

1. There is nothing much to test here (if not all, most of the deprecated warnings were suppressed).
13. Verifying that all the CI checks are successful should be enough.
14. However, if you want to be thorough about reviewing this change, you could quickly smoke test the example `app` and see if it is working as expected.
